AppCenterDataStorage/AppCenterDataStorage/Internal/Util/MSDocumentUtils.h

```diff
@@ -18,6 +20,31 @@ NS_ASSUME_NONNULL_BEGIN
  */
 + (NSDictionary *)documentPayloadWithDocumentId:(NSString *)documentId partition:(NSString *)partition document:(NSDictionary *)document;
 
+/**
+ * Test if a reference is a dictionary that has a key of a given type.
+ *
+ * @param reference The reference to test.
+ * @param key The key to look for in the dictionary reference.
+ * @param keyType The expected key type.
+ */
++ (BOOL)isReferenceDictionaryWithKey:(id _Nullable)reference key:(NSString *)key keyType:(Class)keyType;
```

Comment:
Would be particularly useful if we make this one a category over NSObject. We can pair on it. 
Basically importing this category you would be able to call `[myObject isReferenceDictionaryWithKey...]`.

Reply:
A bit overkill, at least we tried.